docs/docs-content/troubleshooting/palette-dev-engine.md

```diff
@@ -147,3 +147,61 @@ limits for a virtual cluster, and use the host cluster's default resource limits
    ```
 
 4. Repeat step one through three for each host cluster that makes up the Cluster Group.
+
+## Scenario - Adjust Virtual Clusters Limits Before Palette Upgrades
+
+Palette upgrades on K3s virtual clusters upgrades may get stuck if the cluster does not have enough resources to
+accommodate additional pods. Ensure that your cluster has 1 CPU, 1 GiB of memory, and 1 GiB storage of free resources
+before commencing an upgrade.
+
+### Debug Steps
+
+1. Log in to [Palette](https://console.spectrocloud.com).
+
+2. Select **Cluster Groups** from the left **Main Menu**. The list of cluster groups appears.
+
+3. Select the **Virtual Clusters** tab. Then, select one of your virtual clusters.
+
+4. Click **Settings**. Then, select **Cluster Settings**. The **Settings** pane appears.
+
+5. Select the **Cluster Size** tab. Make a note of the CPU, memory and storage allocation indicated in their respective
+   fields. Then, close the **Settings** pane.
+
+6. Click on the host cluster link in the **Host Cluster** field. The cluster **Overview** tab appears.
+
+7. Download the **kubeconfig** file for your cluster and connect to it using
+   [kubectl CLI](https://kubernetes.io/docs/reference/kubectl/). Refer to the
+   [Access Cluster with CLI](../clusters/cluster-management/palette-webctl.md#access-cluster-with-cli) section for
+   further guidance.
+
```

Comment:
We are missing the step where the user opens up the terminal and navigates to it.